src/lua/ext_string.lua

```diff
@@ -0,0 +1,268 @@
+local fun = require('fun')
+
+-- Internal version without checks (assume, that input data is OK)
+local function string_basesplit_iter(inp, sep, max)
+    local last    = 1
+    local pos     = 0
+    local max     = max or 100000000000000
+    local done    = false
+    local matcher = inp:gmatch('(.-)' .. sep .. '()')
```

Comment:
This implementation is extremely inefficient because both gmatch and closure are extremely slow. Can we rewrite it using strstr()/memmem() instead?